fedimint-core/src/config.rs

```diff
@@ -198,16 +198,30 @@ impl ClientConfig {
     }
 
     /// Get the value of a given meta field
-    pub fn meta<V: serde::de::DeserializeOwned>(
+    pub fn meta<V: serde::de::DeserializeOwned + 'static>(
         &self,
         key: &str,
     ) -> Result<Option<V>, anyhow::Error> {
         let Some(str_value) = self.global.meta.get(key) else {
             return Ok(None);
         };
-        serde_json::from_str(str_value)
+        let res = serde_json::from_str(str_value)
             .map(Some)
-            .context(format!("Decoding meta field '{key}' failed"))
+            .context(format!("Decoding meta field '{key}' failed"));
+
+        // In the past we encoded some string fields as "just a string" without quotes,
+        // this code ensures that old meta values still parse since config is hard to
+        // change
+        if res.is_err() && std::any::TypeId::of::<V>() == std::any::TypeId::of::<String>() {
+            let string_ret = Box::new(str_value.clone());
+            let ret: Box<V> = unsafe {
+                // We can transmute a String to V because we know that V==String
+                std::mem::transmute(string_ret)
```

Comment:
Putting bounds here explicitly would make paranoid people feel a bit better about it.